.github/workflows/CrateVersionUpdater.yml

```
@@ -0,0 +1,133 @@
+# @file CrateVersionUpdater.yml
+#
+# A Github workflow that creates or updates a pull request to update the crate version numbers of local crates to match
+# the version specified in the current draft release tag.
+#
+##
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+##
+on:
+  workflow_call:
+    inputs:
+      ignore-prefix:
+        description: 'The prefix to ignore in the release tag (e.g., "patina-v").'
+        required: false
+        default: ''
+        type: string
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+      actions: read
+
+    steps:
+      - name: Validate on main
+        if: github.ref != 'refs/heads/main'
+        run: |
+          echo "This workflow can only be triggered on the main branch."
+          exit 1
+      - name: ✅ Checkout Repository ✅
+        uses: actions/checkout@v5
+
+      - name: 🛠️ Download Rust Tools 🛠️
+        uses: OpenDevicePartnership/patina-devops/.github/actions/rust-tool-cache@main
+
+      - name: Generate Token
+        id: app-token
+        uses: actions/create-github-app-token@v2
+        with:
+          app-id: ${{ secrets.PATINA_AUTOMATION_APPLICATION_ID }}
+          private-key: ${{ secrets.PATINA_AUTOMATION_APPLICATION_PRIVATE_KEY }}
+          owner: ${{ github.repository_owner }}
+
+      - name: Get Release Tag
+        id: release_tag
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ steps.app-token.outputs.token }}
+          script: |
+            const prefix = `${{ inputs.ignore-prefix }}`;
+
+            const { data: releases } = await github.rest.repos.listReleases({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            });
+
+            const draft = releases.find(r => r.draft);
+            if (!draft) {
+              core.setFailed("No draft release found.");
+              return;
+            }
+
+            let tag = draft.tag_name;
+            if (prefix && tag.startsWith(prefix)) {
+              tag = tag.slice(prefix.length);
+            }
+
+            core.info(`Release Tag: ${tag}`);
+            core.setOutput("tag", tag);
+
+      - name: Update git credentials
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+
+      - name: Checkout New Branch
+        run: |
+          git checkout -b "github-actions/release-version-update-${{ steps.release_tag.outputs.tag }}"
+      
+      - name: Run Cargo Release to Update Versions
+        run: |
+          cargo release version ${{ steps.release_tag.outputs.tag }} --execute --no-confirm
+      
+      - name: Push Changes
+        run: |
+          git add .

```

If this is only meant to update Cargo.toml files, I'd constrain that here to reduce the likelihood of some temporary file being included.


```suggestion
          git add **/Cargo.toml
```